Sources/Views/ALKChatCell.swift

```diff
@@ -339,6 +344,18 @@ public final class ALKChatCell: SwipeTableViewCell, Localizable {
 
             onlineStatusView.isHidden = !contact.connected
         }
+        
+        var currentTags = [String]()
+        if let metaData = viewModel.messageMetadata,
+              let tagsData = (metaData["KM_TAGS"] as? String)?.data(using: .utf8),
+           let tagsArray = try? JSONDecoder().decode([KMTags].self, from: tagsData)
+        {
+            for tag in tagsArray {
+                currentTags.append(tag.name)
+            }
+        }
+        customTagView.tagNames = currentTags
+        contentView.layoutIfNeeded()
```

Comment:
This function is becoming too big. Can you separate it into small logical blocks (functions)?

Reply:
created 2 private functions & delegated exceptions